src/pharmpy/modeling/odes.py

```diff
@@ -237,13 +243,15 @@ def add_bioavailability(model: Model, add_parameter: bool = True):
     if odes is None:
         raise ValueError(f'Model {model.name} has no ODE system')
 
-    dose_comp = odes.dosing_compartment[0]
+    dose_comp = odes.first_dosing_compartment
     bio = dose_comp.bioavailability
 
     if isinstance(bio, sympy.Number):
         # Bio not defined
         if add_parameter:
-            model, bio_symb = _add_parameter(model, 'BIO', init=float(bio))
+            model, bio_symb = _add_parameter(
+                model, 'BIO', init=float(init), lower=float(lower), upper=float(upper)
```

Comment:
No need to explicitly convert to `float` I think, in `add_population_parameter` it will use the `Parameter.create()` method.

Reply:
Ah okay I was a bit hasty, it seems we do this conversion for `upper` and `lower`, but not `init`:

https://github.com/pharmpy/pharmpy/blob/87bfb030b0456978061624d8cfabf5840891d4e6/src/pharmpy/model/parameters.py#L67-L74

Would it make sense to do so here as well? Or is there a particular reason it's not done for `init` @rikardn? Was thinking it would be nice to not have to think about it when using the `Parameter.create()` method.